### ISSUE TITLE
Questions about bdd100k data preprocess mentioned in the paper

### ISSUE THREAD

**Issue by nieyan** [cardwing/Codes-for-Lane-Detection#127]
**The lanes of BDD100K are labelled by two lines. Training the networks using the provided labels is tricky. Therefore, based on these two lines, we calculate the center lines as new targets**
- 1. How to pair disorder lanes in the json file ?  
- 2. Do you calculate the center lines of the testing set too? 

Thanks a lot.

**Comment by cardwing** [comment]
BDD100K just records the coordinates of the lane points as annotations. Hence, to calculate the centerline, you just average the coordinates of two neighboring points and the averaged value is just the coordinate of the lane point for the centerline. For the testing set, I also do the same operation. And I will upload the scripts of processing the annotations of BDD100K soon.

**Comment by nieyan** [comment]
That's great! By the way, 8 pixels dense targets w.r.t 360x640 image size?

**Comment by cardwing** [comment]
We dilate ground-truth lanes of the training set of BDD100K as 8 pixels to provide denser targets while
keeping these of testing set unchanged (2 pixels).

**Comment by nieyan** [comment]
I get it , that would be 4 pixels for training set and 1 pixels for testing set while resize to 360x640.  What's more, I am confused about the difference between L_seg and L_exist loss. Do the L_seg have more num classes than  L_exist such as single dash line, double full line.

**Comment by cardwing** [comment]
I only perform binary segmentation in BDD100K since there are so many lanes in the image. Hence, the lane existence prediction branch is removed for BDD100K.

**Comment by ghost** [comment]
> BDD100K just records the coordinates of the lane points as annotations. Hence, to calculate the centerline, you just average the coordinates of two neighboring points and the averaged value is just the coordinate of the lane point for the centerline. For the testing set, I also do the same operation. And I will upload the scripts of processing the annotations of BDD100K soon.

Can you share the implementation details of preprocessing the lanes as the process of combining two lines to form a centerline is not clear?
